File: Core/DotNet/MediaHatchCore/TvShowEpisode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace MediaHatchCore
{
    public class TvShowEpisode : Media
    {
		[JsonProperty("seasonNb")]
        public int SeasonNb { get; set; }

		[JsonProperty("episodeNb")]
        public int[] EpisodeNb { get; set; }

		public override string FullName(string name = null)
		{
			string episodeNumbers = string.Empty;
			for (int i = 0; i < this.EpisodeNb.Length; i++)
				episodeNumbers += $"E{StringifyNumber(this.EpisodeNb[i])}";
			return $"{name ?? Name} S{StringifyNumber(SeasonNb)}{episodeNumbers}";
		}

		private string StringifyNumber(int number)
		{
			return number < 10 ? $"0{number}" : number.ToString();
		}
    }
}

File: Core/media-hatch-core/src/tv-show-episode.ts
import { Media } from './media';
import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject()
export class TvShowEpisode extends Media {
	public static typeName = 'MediaHatchCore.TvShowEpisode';

	@JsonProperty('seasonNb')
	public seasonNb: number = undefined;

	@JsonProperty('episodeNb', [Number])
	public episodeNb: number[] = undefined;

	public get fullName() {
		let episodeNumbers = '';
		for (let i = 0; i < this.episodeNb.length; i++) {
			episodeNumbers += `E${this.stringifyNumber(this.episodeNb[i])}`;
		}
		return `${this.name} S${this.stringifyNumber(this.seasonNb)}${episodeNumbers}`;
	}

	constructor() {
		super(TvShowEpisode.typeName);
		this.episodeNb = new Array<number>();
	}

	private stringifyNumber(number: Number): string {
		return number < 10 ? `0${number}` : number.toString();
	}
}
